src/main/java/edu/ie3/datamodel/io/source/sql/SqlWeatherSource.java

```diff
@@ -146,6 +159,40 @@ public SqlWeatherSource(
     return Optional.of(timeBasedValues.get(0));
   }
 
+  /**
+   * Determine the corresponding the database column name based on the provided factory field
+   * parameter name. Needed to support camel as well as snake case database column names.
+   *
+   * @param factoryColumnName the name of the field parameter set in the entity factory
+   * @param connector the sql connector of this source
+   * @param weatherTableName the table name where the weather is stored
+   * @return the column name that corresponds to the provided field parameter or an empty optional
+   *     if no matching column can be found
+   */
+  private Optional<String> dbColumnName(
+      String factoryColumnName, SqlConnector connector, String weatherTableName) {
+
+    // get the column names from the database
+    Optional<String> dbColumnName = Optional.empty();
+    try {
+      ResultSet rs =
+          connector.getConnection().getMetaData().getColumns(null, null, weatherTableName, null);
+
+      while (rs.next()) {
+        String databaseColumnName = rs.getString("COLUMN_NAME");
+        if (StringUtils.snakeCaseToCamelCase(databaseColumnName)
+            .equalsIgnoreCase(factoryColumnName)) {
+          dbColumnName = Optional.of(databaseColumnName);
+          break;
+        }
+      }
+
+    } catch (SQLException throwables) {
+      throwables.printStackTrace();
```

Comment:
Is there a reason to not log this exception properly (cf. vulnerability detected by SonarQube)?

Reply:
nope - looks like I missed adapting the auto-generated code for try-catch